tests/PySys/apt_plugin/apt_prepare/run.py

```diff
@@ -0,0 +1,38 @@
+import sys
+import os
+import time
+
+sys.path.append("apt_plugin")
+from environment_apt_plugin import AptPlugin
+
+"""
+Validate apt plugin prepare
+(As far as currently possible)
```

Comment:
This test should be rewritten from a user perspective. The point of a `prepare` is to ensure that a given version of a software is available.

Hence, I would go along the lines:
1) Use a docker image where version x.y.z of module bar is not available yet.
2) Check that this version cannot be directly install with apt.
3) Check that this version can be install with the plugin (that invokes an update behind the scene).

Reply:
I like the docker version, too. Albeit, I'm not fully convinced what we gain by really making sure that we have new versions in the apt-cache. The cache will take a long time to update and we would need to create an old Docker image on purpose. Then it would be necessary to make sure that the tests run in a specific order which makes them dependent on each other.

Reply:
I played a bit with deleting /var/cache/apt/pkgcache.bin but that did not really work out as expected